sdlib/d/gc/emap.d

```diff
@@ -27,9 +27,15 @@ public:
 		return leaf is null ? PageDescriptor(0) : leaf.load();
 	}
 
+	bool extend(void* address, size_t size) shared {
+		auto lastPd = lookup(address - PageSize);
+		assert(lastPd.extent !is null, "Nothing mapped before address!");
```

Comment:
You don't want to do a lookup here, when you have all that you need in the extent to compute the right value.